packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -9108,28 +9108,31 @@ function Adapter(options) {
                     }
                 }
 
-                // auto oStates
-                if (options.states) {
-                    this.getStates('*', null, (err, _states) => {
-                        this.oStates = _states;
-                        this.subscribeStates('*');
-                        if (firstConnection) {
-                            firstConnection = false;
-                            typeof options.ready === 'function' && options.ready();
-                            this.emit('ready');
-                        } else {
-                            typeof options.reconnect === 'function' && options.reconnect();
-                            this.emit('reconnect');
-                        }
+                if (!stopInProgress) {
+                    // auto oStates
+                    if (options.states) {
+                        this.getStates('*', null, (err, _states) => {
+                            this.oStates = _states;
```

Comment:
should we not add the same guard here again or better move to there?

Reply:
Yep after the async stuff